src/Resources/app/storefront/test/filter-category-select.plugin.test.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import FilterCategorySelectPlugin from "../src/js/filter-category-select.plugin";
+import ListingPlugin
+  from "../../../../../../../../vendor/shopware/platform/src/Storefront/Resources/app/storefront/src/plugin/listing/listing.plugin";
+
+describe('filter-category-select.plugin.js', () => {
+  let filterCategorySelectPlugin;
+
+  beforeEach(() => {
+    window.csrf = {
+      enabled: false
+    };
+
+    window.router = [];
+
+    const mockElement = document.createElement('div');
+
+    const cmsElementProductListingWrapper = document.createElement('div');
+    cmsElementProductListingWrapper.classList.add('cms-element-product-listing-wrapper');
+
+    const mockElementSpan = document.createElement('span');
+    mockElementSpan.classList.add('filter-multi-select-count');
+
+    const checkboxSelector = document.createElement('input');
+    checkboxSelector.classList.add('filter-category-select-checkbox');
+
+    const mockElementButton = document.createElement('button');
+    mockElementButton.classList.add('filter-panel-item-toggle');
+
+    mockElement.appendChild(cmsElementProductListingWrapper);
+    mockElement.appendChild(mockElementButton);
+    mockElement.appendChild(mockElementSpan);
+    mockElement.appendChild(checkboxSelector);
```

Comment:
Can `appendChild` be chained?

Reply:
It can be chained, but the return value is the appended child and not the `mockElement`.

https://developer.mozilla.org/en-US/docs/Web/API/Node/appendChild

![image](https://user-images.githubusercontent.com/20555112/145008423-273ea597-0733-4407-a548-dce6023f282c.png)